app/src/main/java/com/example/habitsmasher/FollowUserDialog.java

```diff
@@ -91,9 +99,17 @@ public void onComplete(@NonNull Task<QuerySnapshot> task) {
                                         Log.d(TAG, "Username exists");
 
                                         // if username exists, update follower/following counts
-                                        String currentUserId = getCurrentUserId();
+                                        String currentUserId = UserDatabaseHelper.getCurrentUser(_context).getId();
+                                        String currentUserUsername = UserDatabaseHelper.getCurrentUser(_context).getUsername();
 
                                         User userToFollow = document.toObject(User.class);
+
+                                        // if the user follows themselves, display error and return
+                                        if (userToFollow.getUsername().equals(currentUserUsername)) {
+                                            displayErrorMessage(CANNOT_FOLLOW_YOURSELF_ERROR);
+                                            return;
+                                        }
```

Comment:
Here is the new check. It's pretty simple, just checking if the username of the user to follow is the **same** as the current user.